lib/features/currency/payin.dart

```diff
@@ -42,7 +43,10 @@ class Payin extends HookWidget {
         : Currency.formatFromString(amount.value);
 
     useEffect(() {
-      WidgetsBinding.instance.addPostFrameCallback((_) => amount.value = '0');
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        amount.value = '0';
+        decimalPaddingHint.value = '';
```

Comment:
Isn't this the `initialValue` set above?

Reply:
this `useEffect` is being used to reset the user input whenever the currency changes. if `amount` is being reset, then `decimalPaddingHint` needs to be reset. for example, if `decimalPaddingHint` is `'.00'` when `currency` is usd, switching to btc should make sure to set `decimalPaddingHint` to `''` instead of displaying `.00` (which is an incorrect hint)

Reply:
Ah totally makes sense. I didn't see the `dependencies` for this in the PR review. Sorry for the confusion!